src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -78,54 +70,16 @@ internal class GrpcWorker : IWorker
 
         public async Task StartAsync(CancellationToken token)
         {
-            var eventStream = _rpcClient.EventStream(cancellationToken: token);
-
-            await SendStartStreamMessageAsync(eventStream.RequestStream);
-
-            _ = StartWriterAsync(eventStream.RequestStream);
-            _ = StartReaderAsync(eventStream.ResponseStream);
+            _workerClient = await _workerClientFactory.StartClientAsync(this, token);
         }
 
-        public Task StopAsync(CancellationToken token)
-        {
-            return Task.CompletedTask;
-        }
+        public Task StopAsync(CancellationToken token) => Task.CompletedTask;
 
-        private async Task SendStartStreamMessageAsync(IClientStreamWriter<StreamingMessage> requestStream)
-        {
-            StartStream str = new StartStream()
-            {
-                WorkerId = _startupOptions.WorkerId
-            };
-
-            StreamingMessage startStream = new StreamingMessage()
-            {
-                StartStream = str
-            };
-
-            await requestStream.WriteAsync(startStream);
-        }
-
-        private async Task StartWriterAsync(IClientStreamWriter<StreamingMessage> requestStream)
-        {
-            await foreach (StreamingMessage rpcWriteMsg in _outputReader.ReadAllAsync())
-            {
-                await requestStream.WriteAsync(rpcWriteMsg);
-            }
-        }
-
-        private async Task StartReaderAsync(IAsyncStreamReader<StreamingMessage> responseStream)
-        {
-            while (await responseStream.MoveNext())
-            {
-                await ProcessRequestAsync(responseStream.Current);
-            }
-        }
-
-        private Task ProcessRequestAsync(StreamingMessage request)
+        Task IMessageProcessor.ProcessMessageAsync(StreamingMessage message)
         {
             // Dispatch and return.
-            Task.Run(() => ProcessRequestCoreAsync(request));
+            Task.Run(() => ProcessRequestCoreAsync(message));
```

Comment:
This seems slightly odd to me: the contract is `Task` returning, but this will fire and forget regardless. In these cases, shouldn't the caller decide if they want to fire & forget vs wait on the completion?

Of course, I may be missing some context here. But maybe there is a way we can shape this contract to line up with the implementation a bit more? Might be as simple as `Task ProcessMessageAsync` -> `ValueTask PostMessageAsync`.

Additionally - is `Task.Run()` necessary here? Or would a:

``` CSharp
_ = ProcessRequestCoreAsync(message);
```

be good enough?

Reply:
This was pre-existing code, but yes, we should be able to use a discard here, so I'll update.

I think the suggested interface change makes sense, but let me follow up with a different PR so I'm not increasing the scope of the change here (as that is also pre-existing).